data/filedefs/filetypes.Arduino.conf

```diff
@@ -0,0 +1,44 @@
+[styling=C]
+
+
+[keywords]
+primary=and and_eq asm auto bitand bitor bool boolean break byte case catch char class compl const const_cast constexpr continue decltype default delete do double dynamic_cast else enum explicit extern false final float for friend goto if inline int int8_t int16_t int32_t int64_t long mutable namespace new noexcept not not_eq nullptr operator or or_eq override private protected ptrdiff_t public register reinterpret_cast return short signed sizeof size_t static static_assert static_cast struct switch template this throw true try typedef typeid typename uint8_t uint16_t uint32_t uint64_t union unsigned using virtual void volatile while xor xor_eq
+secondary=abs analogRead analogReadResolution analogReference analogWrite analogWriteResolution attachInterrupt bit bitClear bitRead bitSet bitWrite constrain cos delay delayMicroseconds detachInterrupt digitalRead digitalWrite EEPROM Ethernet F highByte interrupts isAlpha isAlphaNumeric isAscii isControl isDigit isGraph isHexadecimalDigit isLowerCase isPrintable isPunct isSpace isUpperCase isWhitespace Keyboard LiquidCrystal loop lowByte map max micros millis min Mouse noInterrupts noTone pinMode pow PROGMEM pulseIn random randomSeed SD Serial Servo setup shiftIn shiftOut sin SoftwareSerial SPI sqrt Stepper tan tone Wire word yield
+
+
+[lexer_properties=C]
+
+
+[settings]
+lexer_filetype=C
+tag_parser=C
+extension=ino
+
+# context action command (please see Geany's main documentation for details)
+context_action_cmd=xdg-open "https://www.arduino.cc/en/Reference/%s"
+
+
+[indentation]
+#width=2
+# 0 is spaces, 1 is tabs, 2 is tab & spaces
+#type=0
+
+
+# Use this reference for making up Arduino command lines:
+# https://github.com/arduino/Arduino/blob/master/build/shared/manpage.adoc
+[build-menu]
+FT_00_LB=_Build
+FT_00_CM=arduino --verify "%d/%f" --board arduino:avr:uno
+FT_00_WD=
+
+FT_01_LB=Build (_Verbose)
+FT_01_CM=arduino --verify --verbose-build "%d/%f" --board arduino:avr:uno
```

Comment:
The Uno is the "standard" Arduino nowadays, definitely the most common one. Of course I would love to have a board selection menu, but I guess we need a plugin rather than a filetype, for that. In fact, I am actually considering the possibility of writing an Arduino plugin.

That option can only be left out if you want to compile for the latest board that you selected in the Arduino IDE, but I think it is more practical to override it by customizing the command line in the project settings, rather than having to open the Arduino IDE and change the board there. If I have to do that, at that point I can just hit the Compile button in the IDE to do the build.